### PR TITLE
fix: --all bypass .gitignore rules (#285)

### PR DESCRIPTION
**Pull request by aitva**
I was very surprised when I first tried Glow since I could not find any of my files. It took me some time to stumble upon #285 and discover that Glow was ignoring my entire home directory because of a `.gitignore`.

This commit fixes the issue by adding a  `findFiles` function can perform a search without using the rules in `.gitignore`. The function reuse the signature from `gitcha.FindFilesExcept` and is called when Glow is invoked with the `--all` flag.

### CODE REVIEW

**Comment by uncomfyhalomacro** [comment]
> I was very surprised when I first tried Glow since I could not find any of my files. It took me some time to stumble upon #285 and discover that Glow was ignoring my entire home directory because of a `.gitignore`.
> 
> This commit fixes the issue by adding a `findFiles` function can perform a search without using the rules in `.gitignore`. The function reuse the signature from `gitcha.FindFilesExcept` and is called when Glow is invoked with the `--all` flag.

seems no one is checking this PR. weird. is this project abandoned?

File: ui/ui.go [hunk 1]
```diff
@@ -505,12 +505,13 @@ func findLocalFiles(m model) tea.Cmd {
...
-		if !m.common.cfg.ShowAllFiles {
-			ignore = ignorePatterns(m)
+		var ch chan searchResult
+		if m.common.cfg.ShowAllFiles {
+			ch, err = findFiles(cwd, markdownExtensions)
+		} else {
+			ch, err = findFilesExcept(cwd, markdownExtensions, ignorePatterns(m))
 		}
 
-		ch, err := gitcha.FindFilesExcept(cwd, markdownExtensions, ignore)
```
Comment (caarlos0) [inline ui/ui.go:513]: couldn't we pass an empty `ignore` list when using `--all`?

**Comment by maaslalani** [comment]
Hey @aitva @uncomfyhalomacro, I just made this PR in `gitcha` to solve this issue, we can then simply use `FindAllFiles[Except]` to bypass the `.gitignore`:

https://github.com/muesli/gitcha/pull/4

**Comment by uncomfyhalomacro** [comment]
> Thank you @maaslalani, I will update the PR when [muesli/gitcha#4](https://github.com/muesli/gitcha/pull/4) is merged.

Seems it got merged :D

**Comment by aitva** [comment]
I have updated my merge request to use the latest version of gitcha. :+1:

**Comment by karpfediem** [comment]
Hi, what is the status on this? It looks ready to be merged.
The issue https://github.com/charmbracelet/glow/issues/285 still exists. Glow is unusable with certain projects using .gitignore

File: ui/ui.go [hunk 2]
```diff
@@ -331,12 +331,15 @@ func findLocalFiles(m commonModel) tea.Cmd {
 		}
 
 		log.Debug("local directory is", "cwd", cwd)
-		var ignore []string
-		if !m.cfg.ShowAllFiles {
-			ignore = ignorePatterns(m)
+
+		// Switch between FindFiles and FindAllFiles to bypass .gitignore rules
+		var ch chan gitcha.SearchResult
+		if m.common.cfg.ShowAllFiles {
```
Comment (caarlos0) [inline ui/ui.go:337]:
```suggestion
		if m.cfg.ShowAllFiles {
```

**Comment by caarlos0** [comment]
thanks for the PR, and sorry for the delay!

**Pull request merged**